primitives/src/vid/namespace.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Trait for namespace functionality in Verifiable Information Retrieval (VID).
+
+use super::{VidResult, VidScheme};
+
+/// Namespace functionality for [`VidScheme`].
+pub trait Namespacer: VidScheme {
+    /// data proof
+    type DataProof;
+
+    /// Compute a proof for `payload` for data index range `start..start+len-1`.
+    ///
+    /// See TODO in `namespace_verify` on `payload`.
+    fn data_proof(
+        &self,
+        payload: &Self::Payload,
+        start: usize,
+        len: usize,
+    ) -> VidResult<Self::DataProof>;
+
+    /// Verify a proof for `payload` for data index range `start..start+len-1`.
+    ///
+    /// See TODO in `namespace_verify` on `payload`.
+    fn data_verify(
+        &self,
+        payload: &Self::Payload,
+        start: usize,
+        len: usize,
+        commit: &Self::Commit,
+        common: &Self::Common,
```

Comment:
Are `commit` and `common` always together? Would it make sense to just put all the common stuff in `commit`?

Reply:
`commit` is supposed to be an extremely small item (like a 32-byte hash).

`common` is metadata sent to all storage nodes such as KZG commitments to all the polynomials in a block payload. It's size is not bounded. I view it more as part of a proof to be verified against `commit` than as part of `commit` itself.

Reply:
Gotcha